### PR TITLE
#445 - update matrix-auth demo

### PR DESCRIPTION
**Pull request by jonbrohauge**
Updates the example yaml configuration to reflect breaking changes going to 1.0-rc1

### CODE REVIEW

**Comment by x10an14** [comment]
Confirmed to work, with the following caveats! =D
  * there can _max_ be two spaces between `-` and `script:`
  * and you _must_ indent every line after `script:`s line up to (or beyond) the column of the `s` in `script:` itself.

File: demos/global-matrix-auth/README.md [hunk 1]
```diff
@@ -7,12 +7,8 @@ jenkins:
   authorizationStrategy:
     globalMatrix:
       grantedPermissions:
-        - name: "anonymous"
-          permissions:
-            - "Overall/Read"
-        - name: "authenticated"
-          permissions:
-            - "Overall/Administer"
+        - "Overall/Read:anonymous"
```
Comment (casz) [inline demos/global-matrix-auth/README.md]: I don't the the quoting is needed but I guess it is a style preference 😅

Comment (x10an14) [reply demos/global-matrix-auth/README.md]: @casz This is mine atm:
```yaml
  authorizationStrategy:
    globalMatrix:
      grantedPermissions:
        # Necessary for logged-in users to view and run jobs:
        - "Job/Read:authenticated"
        - "View/Read:authenticated"
        - "Overall/Read:authenticated"
        - "Job/Cancel:authenticated"
        - "Job/Build:authenticated"
        # Necessary for admin view (which also includes config):
        - "Overall/Administer:x10an14"
```

**Comment by ewelinawilkosz** [comment]
@x10an14 is your comment related to this PR?

**Pull request merged**